src/core/Akka.Streams.Tests/Dsl/FlowWithContextSpec.cs

```diff
@@ -0,0 +1,80 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="FlowWithContextSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2021 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2021 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Akka.Streams.Dsl;
+using Akka.Streams.TestKit;
+using Akka.TestKit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Streams.Tests.Dsl
+{
+    public class FlowWithContextSpec : AkkaSpec
+    {
+        private ActorMaterializer Materializer { get; }
+
+        public FlowWithContextSpec(ITestOutputHelper helper) : base(helper)
+        {
+            var settings = ActorMaterializerSettings.Create(Sys);
+            Materializer = ActorMaterializer.Create(Sys, settings);
+        }
+
+        [Fact]
+        public void A_FlowWithContext_must_get_created_from_FlowAsFlowWithContext()
+        {
+            var flow = Flow.Create<Message>().Select(m => m.Copy(data: m.Data + "z"));
+            var flowWithContext = flow.AsFlowWithContext<Message, long, Message, long, NotUsed, Message>((m, o) => new Message(m.Data, o), m => m.Offset);
```

Comment:
Why can't the compiler infer the type arguments? This should be just: `flow.AsFlowWithContext((m, o) => new Message(m.Data, o), m => m.Offset);`